receiver/splunkhecreceiver/factory.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecreceiver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+)
+
+// This file implements factory for Splunk HEC receiver.
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "splunk_hec"
+
+	// Default endpoints to bind to.
+	defaultEndpoint = ":8088"
+)
+
+// Factory is the factory for Splunk HEC receiver.
+type Factory struct {
+}
+
+var _ component.ReceiverFactoryOld = (*Factory)(nil)
```

Comment:
Factory old was removed, you can use the receiverhelper and create the new factory type (see signalfx or sapm as example).